src/devices/vdml_imu.c

```diff
@@ -39,8 +41,15 @@ int32_t imu_reset(uint8_t port) {
 	claim_port_i(port - 1, E_DEVICE_IMU);
 	ERROR_IMU_STILL_CALIBRATING(port, device, PROS_ERR);
 	vexDeviceImuReset(device->device_info);
-	// delay for vexos to set calibration flag
-	while(!(vexDeviceImuStatusGet(device->device_info) & E_IMU_STATUS_CALIBRATING)) delay(5);
+	// delay for vexos to set calibration flag, background processing must be called for flag
+	// to be set.
+	uint16_t timeoutCount = 0;
+	do {
+		delay(5);
+		timeoutCount += 5;
+		vexBackgroundProcessing();
+	} while(!(vexDeviceImuStatusGet(device->device_info) & E_IMU_STATUS_CALIBRATING) && timeoutCount < IMU_RESET_TIMEOUT);
```

Comment:
Not sure if it would be? While we do keep track of the previous time before incrementing the timeout count it's just another parameter that does essentially the same thing for the delay function.

Reply:
It also avoids accumulation of extra time, not sure if that could be an issue with `vexBackgroundProcessing()` or not.

Reply:
How does releasing the port mutex before we sleep sound? It'll also handle the case where someone switches out/disconnects the device and we get an updated view of it during vexBackgroundProcessing?